### ISSUE TITLE
question about network input and ae loss

### ISSUE THREAD

**Issue by zhengbowei** [JiahuiYu/generative_inpainting#436]
Dear author:
I have two little questions for you：
1,
In the CoarseNetwork, the network input is:
x = tf.concat([x, ones_x, ones_x*mask], axis=3)

My question is tf.concat "ones_x*mask": 0 indicates the outside known region and 1 indicates the inside missing region.
Didn't it mean that this operation masks the known region and shows the missing region? Is it right？

Thank you for your answer。

2,ae_loss
batch_pos = batch_data / 127.5 - 1.
x1, x2, offset_flow = self.build_inpaint_net(xin, mask, reuse=reuse, training=training, padding=FLAGS.padding)

# local patches
losses['ae_loss'] = FLAGS.l1_loss_alpha * tf.reduce_mean(tf.abs(batch_pos - x1))
losses['ae_loss'] += FLAGS.l1_loss_alpha * tf.reduce_mean(tf.abs(batch_pos - x2))

Here x1 and x2 do not replace the known real region, so the loss here refers to the entire picture of network output as loss. May I ask why we should make loss on the known real region? Is it not the goal of inpainting to only make loss on the missing part?

Thank you for your answer。

**Comment by JiahuiYu** [comment]
1. #252 

2. Applying loss for entire prediction makes network learn better especially for color consistency.
